logic/atrium-logic-common/src/main/kotlin/ch/tutteli/atrium/logic/impl/DefaultIteratorAssertions.kt

```diff
@@ -14,4 +14,17 @@ class DefaultIteratorAssertions : IteratorAssertions {
 
     override fun <E, T : Iterator<E>> hasNotNext(container: AssertionContainer<T>): Assertion =
         container.createDescriptiveAssertion(HAS_NOT, NEXT_ELEMENT) { !it.hasNext() }
+
+    override fun <E, T : Iterator<E>> containsNoDuplicates(container: AssertionContainer<T>): Assertion {
+        val finalList = mutableListOf<E>()
+
+        return container.createDescriptiveAssertion(CONTAINS_DUPLICATES, DUPLICATE_FOUND) {
```

Comment:
IMO the resulting descriptive assertion does not fit into the existing assertions
```
expected that subject: [1, 2, 3, 4, 4]        (java.util.Arrays.ArrayList <1234789>)
◆ contains duplicates: elements in `Iterable`
```
And I think that a user wants to see more here, which elements were duplicates. I image something like:
```
expected that subject: [1, 2, 3, 4, 4]        (java.util.Arrays.ArrayList <1234789>)
◆ has not: duplicate elements 
    ❗❗ following duplicates detected 
       ⚬ index 4: 4        (kotlin.Int <1234789>)
       ⚬ index 5: 5        (kotlin.Int <1234789>)
```
Similar to https://github.com/robstoll/atrium#collection-assertions => all

Reply:
I will try to implement that way